app/src/main/java/org/mozilla/fenix/library/bookmarks/selectfolder/SelectBookmarkFolderFragment.kt

```diff
@@ -126,6 +143,98 @@ class SelectBookmarkFolderFragment : Fragment(), CoroutineScope, AccountObserver
             ?: getManagedEmitter<SignInChange>().onNext(SignInChange.SignedOut)
     }
 
+    private suspend fun virtualDesktopFolder(): BookmarkNode? {
+        val rootNode =
+            requireComponents.core.bookmarksStorage.getTree(BookmarkRoot.Root.id, false) ?: return null
+        return BookmarkNode(
+            type = rootNode.type,
+            guid = rootNode.guid,
+            parentGuid = rootNode.parentGuid,
+            position = rootNode.position,
+            title = rootTitles[rootNode.title],
+            url = rootNode.url,
+            children = rootNode.children
+        )
+    }
+
+    @SuppressWarnings("ReturnCount")
+    private suspend fun withOptionalDesktopFolders(node: BookmarkNode?): BookmarkNode? {
```

Comment:
I only just made this an extension function in BookmarkFragment to ease readability. Is there any reason we need to copy this function rather than simply calling it?

Ultimately, we'll want this in A-C I'd think.